item-impl/src/main/java/com/example/auction/item/impl/CompletionStageUtils.java

```diff
@@ -0,0 +1,43 @@
+package com.example.auction.item.impl;
+
+import akka.Done;
+import com.datastax.driver.core.BoundStatement;
+import com.lightbend.lagom.javadsl.persistence.cassandra.CassandraReadSide;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+final class CompletionStageUtils {
+    private CompletionStageUtils() {
+        throw new Error("no instances");
+    }
+
+    static <R, X extends RuntimeException> Function<Optional<R>, R> throwIfEmpty(Supplier<X> exceptionThrower) {
+        return result -> result.orElseThrow(exceptionThrower);
+    }
+
+    static CompletionStage<List<BoundStatement>> completedStatements(BoundStatement... statements) {
+        return CassandraReadSide.completedStatements(Arrays.asList(statements));
+    }
+
+    static <T> Function<T, Done> accept(Consumer<T> f) {
+        return t -> {
+            f.accept(t);
+            return Done.getInstance();
+        };
+    }
+
+    static CompletionStage<Done> doAll(CompletionStage<?>... stages) {
```

Comment:
Surprised that something like this one isn't in the JDK. There is a `CompletableFuture.allOf` that is similar, but it only works with `CompletableFuture` instances and not arbitrary `CompletionStage` for some reason.

Also, this of course uses `Done` instead of `Void` to conform with the Lagom expectations... why _do_ we use `Done` instead of `Void` anyway? It makes a few things more awkward.